From 1c8621779cd9ab5cd3c5195e806efd51bbf4ab19 Mon Sep 17 00:00:00 2001
From: Radu Pirea <radu-nicolae.pirea@nxp.com>
Date: Tue, 15 Dec 2020 11:42:40 +0200
Subject: [PATCH 36/76] dts: s32-gen1: add pinctrl for i2c, spi, qspi, sdhc,
 usb, gmac, pfe

Add pinctrl configs for peripherals.

Issue: ALB-5668

Signed-off-by: Radu Pirea <radu-nicolae.pirea@nxp.com>
---
 arch/arm/dts/fsl-s32g274a.dtsi        | 161 +++++++++++++++++++++
 arch/arm/dts/fsl-s32g274abluebox3.dts | 126 +++++++++++++++-
 arch/arm/dts/fsl-s32g274aevb.dts      | 189 +++++++++++++++++++++++-
 arch/arm/dts/fsl-s32g274ardb.dtsi     | 153 +++++++++++++++++++-
 arch/arm/dts/fsl-s32r45.dts           | 197 ++++++++++++++++++++++++++
 5 files changed, 823 insertions(+), 3 deletions(-)

diff --git a/arch/arm/dts/fsl-s32g274a.dtsi b/arch/arm/dts/fsl-s32g274a.dtsi
index c6864a6d87..ddd0c614c4 100644
--- a/arch/arm/dts/fsl-s32g274a.dtsi
+++ b/arch/arm/dts/fsl-s32g274a.dtsi
@@ -9,6 +9,7 @@
  */
 #include <dt-bindings/clock/s32g274a-clock.h>
 #include <dt-bindings/clock/s32g274a-scmi-clock.h>
+#include <dt-bindings/pinctrl/s32g274a-pinctrl.h>
 
 /dts-v1/;
 #include "fsl-s32-gen1.dtsi"
@@ -156,6 +157,18 @@
 			"mac1_rx_rgmii", "mac1_tx_rgmii",
 			"mac2_rx_sgmii", "mac2_tx_sgmii",
 			"mac2_rx_rgmii", "mac2_tx_rgmii";
+		pinctrl-0 = <&pinctrl0_pfe0 &pinctrl0_pfe0_mdio &pinctrl1_pfe0
+			&pinctrl1_pfe0_mdio>;
+		pinctrl-1 = <&pinctrl0_pfe0_mdio &pinctrl1_pfe0_mdio>;
+		pinctrl-2 = <&pinctrl0_pfe1 &pinctrl0_pfe1_mdio &pinctrl1_pfe1
+			&pinctrl1_pfe1_mdio>;
+		pinctrl-3 = <&pinctrl0_pfe1_mdio &pinctrl1_pfe1_mdio>;
+		pinctrl-4 = <&pinctrl0_pfe2 &pinctrl0_pfe2_mdio &pinctrl1_pfe2
+			&pinctrl1_pfe2_mdio>;
+		pinctrl-5 = <&pinctrl0_pfe2_mdio &pinctrl1_pfe2_mdio>;
+		pinctrl-names = "pfe0_rgmii", "pfe0_sgmii",
+			"pfe1_rgmii", "pfe1_sgmii",
+			"pfe2_rgmii", "pfe2_sgmii";
 		status = "disabled";
 	};
 
@@ -310,6 +323,9 @@
 	clock-names = "rx_sgmii", "tx_sgmii", "ts_sgmii",
 		      "rx_rgmii", "tx_rgmii", "ts_rgmii",
 		      "axi";
+	pinctrl-0 = <&pinctrl0_gmac0 &pinctrl0_gmac0_mdio>;
+	pinctrl-1 = <&pinctrl0_gmac0_mdio>;
+	pinctrl-names = "gmac_rgmii", "gmac_sgmii";
 };
 
 &qspi {
@@ -357,3 +373,148 @@
 
 	status = "okay";
 };
+
+&pinctrl0 {
+	board_generic_pinctrl0 {
+		pinctrl0_gmac0_mdio: pinctrl0_gmac0_mdio {
+			fsl,pins = <PD12_MSCR_S32G2XX PD12_GMAC0_MDC_CFG
+				    PD13_MSCR_S32G2XX PD13_GMAC0_MDIO_CFG
+				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
+				    >;
+		};
+
+		pinctrl0_gmac0: pinctrl0_gmac0 {
+			fsl,pins = <PE02_MSCR_S32G2XX PE02_GMAC0_TX_CLK_CFG
+				    PE03_MSCR_S32G2XX PE03_GMAC0_TX_EN_CFG
+				    PE04_MSCR_S32G2XX PE04_GMAC0_TX_D0_CFG
+				    PE05_MSCR_S32G2XX PE05_GMAC0_TX_D1_CFG
+				    PE06_MSCR_S32G2XX PE06_GMAC0_TX_D2_CFG
+				    PE07_MSCR_S32G2XX PE07_GMAC0_TX_D3_CFG
+				    PE08_MSCR_S32G2XX PE08_GMAC0_RX_CLK_CFG
+				    PE09_MSCR_S32G2XX PE09_GMAC0_RX_DV_CFG
+				    PE10_MSCR_S32G2XX PE10_GMAC0_RX_D0_CFG
+				    PE11_MSCR_S32G2XX PE11_GMAC0_RX_D1_CFG
+				    PE12_MSCR_S32G2XX PE12_GMAC0_RX_D2_CFG
+				    PE13_MSCR_S32G2XX PE13_GMAC0_RX_D3_CFG
+				    GMAC0_TX_CLK_IMCR PE02_GMAC0_TX_CLK_IN
+				    GMAC0_RX_CLK_IMCR PE08_GMAC0_RX_CLK_IN
+				    GMAC0_RX_DV_IMCR PE09_GMAC0_RX_DV_IN
+				    GMAC0_RX_D0_IMCR PE10_GMAC0_RX_D0_IN
+				    GMAC0_RX_D1_IMCR PE11_GMAC0_RX_D1_IN
+				    GMAC0_RX_D2_IMCR PE12_GMAC0_RX_D2_IN
+				    GMAC0_RX_D3_IMCR PE13_GMAC0_RX_D3_IN
+				    >;
+		};
+
+		pinctrl0_pfe0_mdio: pinctrl0_pfe0_mdio {
+			fsl,pins = <PE15_MSCR_S32G2XX PE15_PFE0_MDIO_CFG
+				    PF02_MSCR_S32G2XX PF02_PFE0_MDC_CFG
+				    >;
+		};
+
+		pinctrl0_pfe0: pinctrl0_pfe0 {
+			fsl,pins = <PE14_MSCR_S32G2XX PE14_PFE0_TX_EN_CFG>;
+		};
+
+		pinctrl0_pfe1_mdio: pinctrl0_pfe1_mdio {
+			fsl,pins = <PD12_MSCR_S32G2XX PD12_PFE1_MDC_CFG
+				    PD13_MSCR_S32G2XX PD13_PFE1_MDIO_CFG
+				    >;
+		};
+
+		pinctrl0_pfe1: pinctrl0_pfe1 {
+			fsl,pins = <PE02_MSCR_S32G2XX PE02_PFE1_TX_CLK_CFG
+				    PE03_MSCR_S32G2XX PE03_PFE1_TX_EN_CFG
+				    PE04_MSCR_S32G2XX PE04_PFE1_TX_D0_CFG
+				    PE05_MSCR_S32G2XX PE05_PFE1_TX_D1_CFG
+				    PE06_MSCR_S32G2XX PE06_PFE1_TX_D2_CFG
+				    PE07_MSCR_S32G2XX PE07_PFE1_TX_D3_CFG
+				    PE08_MSCR_S32G2XX PE08_PFE1_RX_CLK_CFG
+				    PE09_MSCR_S32G2XX PE09_PFE1_RX_DV_CFG
+				    PE10_MSCR_S32G2XX PE10_PFE1_RX_D0_CFG
+				    PE11_MSCR_S32G2XX PE11_PFE1_RX_D1_CFG
+				    PE12_MSCR_S32G2XX PE12_PFE1_RX_D2_CFG
+				    PE13_MSCR_S32G2XX PE13_PFE1_RX_D3_CFG
+				    >;
+		};
+
+		pinctrl0_pfe2_mdio: pinctrl0_pfe2_mdio {
+			fsl,pins = <PE15_MSCR_S32G2XX PE15_PFE2_MDIO_CFG
+				    PF02_MSCR_S32G2XX PF02_PFE2_MDC_CFG
+				    >;
+		};
+
+		pinctrl0_pfe2: pinctrl0_pfe2 {
+			fsl,pins = <PE14_MSCR_S32G2XX PE14_PFE2_TX_EN_CFG>;
+		};
+	};
+};
+
+&pinctrl1 {
+	board_generic_pinctrl1 {
+		pinctrl1_pfe0_mdio: pinctrl1_pfe0_mdio {
+			fsl,pins = <PFE0_MDIO_IMCR PE15_PFE0_MDIO_IN>;
+		};
+
+		pinctrl1_pfe0: pinctrl1_pfe0 {
+			fsl,pins = <PH01_MSCR_S32G2XX PH01_PFE0_TX_D1_CFG
+				    PH02_MSCR_S32G2XX PH02_PFE0_TX_D2_CFG
+				    PH03_MSCR_S32G2XX PH03_PFE0_TX_D3_CFG
+				    PH04_MSCR_S32G2XX PH04_PFE0_RX_CLK_CFG
+				    PH05_MSCR_S32G2XX PH05_PFE0_RX_DV_CFG
+				    PH06_MSCR_S32G2XX PH06_PFE0_RX_D0_CFG
+				    PH07_MSCR_S32G2XX PH07_PFE0_RX_D1_CFG
+				    PH08_MSCR_S32G2XX PH08_PFE0_RX_D2_CFG
+				    PH09_MSCR_S32G2XX PH09_PFE0_RX_D3_CFG
+				    PH10_MSCR_S32G2XX PH10_PFE0_TX_CLK_CFG
+				    PJ00_MSCR_S32G2XX PJ00_PFE0_TX_D0_CFG
+				    PFE0_RX_CLK_IMCR PH04_PFE0_RX_CLK_IN
+				    PFE0_RX_DV_IMCR PH05_PFE0_RX_DV_IN
+				    PFE0_RX_D0_IMCR PH06_PFE0_RX_D0_IN
+				    PFE0_RX_D1_IMCR PH07_PFE0_RX_D1_IN
+				    PFE0_RX_D2_IMCR PH08_PFE0_RX_D2_IN
+				    PFE0_RX_D3_IMCR PH09_PFE0_RX_D3_IN
+				    >;
+		};
+
+		pinctrl1_pfe1_mdio: pinctrl1_pfe1_mdio {
+			fsl,pins = <PFE1_MDIO_IMCR PD13_PFE1_MDIO_IN>;
+		};
+
+		pinctrl1_pfe1: pinctrl1_pfe1 {
+			fsl,pins = <PFE1_RX_CLK_IMCR PE08_PFE1_RX_CLK_IN
+				    PFE1_RX_DV_IMCR PE09_PFE1_RX_DV_IN
+				    PFE1_RX_D0_IMCR PE10_PFE1_RX_D0_IN
+				    PFE1_RX_D1_IMCR PE11_PFE1_RX_D1_IN
+				    PFE1_RX_D2_IMCR PE12_PFE1_RX_D2_IN
+				    PFE1_RX_D3_IMCR PE13_PFE1_RX_D3_IN
+				    >;
+		};
+
+		pinctrl1_pfe2_mdio: pinctrl1_pfe2_mdio {
+			fsl,pins = <PFE2_MDIO_IMCR PE15_PFE2_MDIO_IN
+				    >;
+		};
+
+		pinctrl1_pfe2: pinctrl1_pfe2 {
+			fsl,pins = <PH01_MSCR_S32G2XX PH01_PFE2_TX_D1_CFG
+				    PH02_MSCR_S32G2XX PH02_PFE2_TX_D2_CFG
+				    PH03_MSCR_S32G2XX PH03_PFE2_TX_D3_CFG
+				    PH04_MSCR_S32G2XX PH04_PFE2_RX_CLK_CFG
+				    PH05_MSCR_S32G2XX PH05_PFE2_RX_DV_CFG
+				    PH06_MSCR_S32G2XX PH06_PFE2_RX_D0_CFG
+				    PH07_MSCR_S32G2XX PH07_PFE2_RX_D1_CFG
+				    PH08_MSCR_S32G2XX PH08_PFE2_RX_D2_CFG
+				    PH09_MSCR_S32G2XX PH09_PFE2_RX_D3_CFG
+				    PH10_MSCR_S32G2XX PH10_PFE2_TX_CLK_CFG
+				    PJ00_MSCR_S32G2XX PJ00_PFE2_TX_D0_CFG
+				    PFE2_RX_CLK_IMCR PH04_PFE2_RX_CLK_IN
+				    PFE2_RX_DV_IMCR PH05_PFE2_RX_DV_IN
+				    PFE2_RX_D0_IMCR PH06_PFE2_RX_D0_IN
+				    PFE2_RX_D1_IMCR PH07_PFE2_RX_D1_IN
+				    PFE2_RX_D2_IMCR PH08_PFE2_RX_D2_IN
+				    PFE2_RX_D3_IMCR PH09_PFE2_RX_D3_IN
+				    >;
+		};
+	};
+};
diff --git a/arch/arm/dts/fsl-s32g274abluebox3.dts b/arch/arm/dts/fsl-s32g274abluebox3.dts
index 80520fc79f..ea517ac98d 100644
--- a/arch/arm/dts/fsl-s32g274abluebox3.dts
+++ b/arch/arm/dts/fsl-s32g274abluebox3.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright 2020 NXP
+ * Copyright 2021 NXP
  *
  */
 
@@ -28,14 +28,23 @@
 };
 
 &dspi0 {
+	pinctrl-0 = <&pinctrl0_dspi0 &pinctrl1_dspi0>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &dspi1 {
+	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &usdhc0 {
+	/* By default, sd0 pins are able to work at 100Mhz and 200Mhz */
+	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-1 = <>;
+	pinctrl-2 = <>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	status = "okay";
 	bus-width = <4>;
 };
@@ -50,6 +59,8 @@
 
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c0>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	eeprom@50 {
@@ -71,16 +82,22 @@
 
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &i2c4 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	vr5510 {
@@ -96,3 +113,110 @@
 	};
 
 };
+
+&pinctrl0 {
+	board_pinctrl0 {
+		pinctrl0_dspi0: pinctrl0_dspi0 {
+			fsl,pins = <PA13_MSCR_S32G2XX PA13_SPI0_SCK_CFG
+				    PA14_MSCR_S32G2XX PA14_SPI0_SIN_CFG
+				    PA15_MSCR_S32G2XX PA15_SPI0_SOUT_CFG
+				    PB09_MSCR_S32G2XX PB09_SPI0_CS1_CFG
+				    PB10_MSCR_S32G2XX PB10_SPI0_CS2_CFG
+				    >;
+		};
+
+		pinctrl0_dspi1: pinctrl0_dspi1 {
+			fsl,pins = <PA06_MSCR_S32G2XX PA06_SPI1_SOUT_CFG
+				    PA07_MSCR_S32G2XX PA07_SPI1_CS0_CFG
+				    PA08_MSCR_S32G2XX PA08_SPI1_SCK_CFG
+				    PE04_MSCR_S32G2XX PE04_SPI1_CS3_CFG
+				    PF15_MSCR_S32G2XX PF15_SPI1_SIN_CFG
+				    >;
+		};
+
+		pinctrl0_i2c0: pinctrl0_i2c0 {
+			fsl,pins = <PB00_MSCR_S32G2XX PB00_I2C0_SDA_CFG
+				    PB01_MSCR_S32G2XX PB01_I2C0_SCL_CFG
+				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
+				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
+				    >;
+		};
+
+		pinctrl0_i2c1: pinctrl0_i2c1 {
+			fsl,pins = <PB03_MSCR_S32G2XX PB03_I2C1_SCL_CFG
+				    PB04_MSCR_S32G2XX PB04_I2C1_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_i2c2: pinctrl0_i2c2 {
+			fsl,pins = <PB05_MSCR_S32G2XX PB05_I2C2_SCL_CFG
+				    PB06_MSCR_S32G2XX PB06_I2C2_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_i2c4: pinctrl0_i2c4 {
+			fsl,pins = <PC01_MSCR_S32G2XX PC01_I2C4_SDA_CFG
+				    PC02_MSCR_S32G2XX PC02_I2C4_SCL_CFG
+				    >;
+		};
+
+		pinctrl0_sd0: pinctrl0_sd0 {
+			fsl,pins = <PC14_MSCR_S32G2XX PC14_SD0_CLK_CFG
+				    PC15_MSCR_S32G2XX PC15_SD0_CMD_CFG
+				    PD00_MSCR_S32G2XX PD00_SD0_D0_CFG
+				    PD01_MSCR_S32G2XX PD01_SD0_D1_CFG
+				    PD02_MSCR_S32G2XX PD02_SD0_D2_CFG
+				    PD03_MSCR_S32G2XX PD03_SD0_D3_CFG
+				    PD04_MSCR_S32G2XX PD04_SD0_D4_CFG
+				    PD05_MSCR_S32G2XX PD05_SD0_D5_CFG
+				    PD06_MSCR_S32G2XX PD06_SD0_D6_CFG
+				    PD07_MSCR_S32G2XX PD07_SD0_D7_CFG
+				    PD08_MSCR_S32G2XX PD08_SD0_RST_CFG
+				    PD09_MSCR_S32G2XX PD09_SD0_VSELECT_CFG
+				    PD10_MSCR_S32G2XX PD10_SD0_DQS_CFG
+				    SD0_CMD_IMCR PC15_SD0_CMD_IN
+				    SD0_D0_IMCR PD00_SD0_D0_IN
+				    SD0_D1_IMCR PD01_SD0_D1_IN
+				    SD0_D2_IMCR PD02_SD0_D2_IN
+				    SD0_D3_IMCR PD03_SD0_D3_IN
+				    SD0_D4_IMCR PD04_SD0_D4_IN
+				    SD0_D5_IMCR PD05_SD0_D5_IN
+				    SD0_D6_IMCR PD06_SD0_D6_IN
+				    SD0_D7_IMCR PD07_SD0_D7_IN
+				    SD0_DQS_IMCR PD10_SD0_DQS_IN
+				    >;
+		};
+	};
+};
+
+&pinctrl1 {
+	board_pinctrl1 {
+		pinctrl1_dspi0: pinctrl1_dspi0 {
+			fsl,pins = <DSPI0_SIN_IMCR PA14_SPI0_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi1: pinctrl1_dspi1 {
+			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
+				    >;
+		};
+
+		pinctrl1_i2c1: pinctrl1_i2c1 {
+			fsl,pins = <I2C1_SCL_IMCR PB03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c2: pinctrl1_i2c2 {
+			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c4: pinctrl1_i2c4 {
+			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
+				    >;
+		};
+	};
+};
diff --git a/arch/arm/dts/fsl-s32g274aevb.dts b/arch/arm/dts/fsl-s32g274aevb.dts
index 8790f327ad..958cffd3c2 100644
--- a/arch/arm/dts/fsl-s32g274aevb.dts
+++ b/arch/arm/dts/fsl-s32g274aevb.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 /*
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  */
 
 /dts-v1/;
@@ -43,6 +43,8 @@
 	usbotg: usb@44064000 {
 		compatible = "fsl,imx27-usb";
 		reg = <0x0 0x44064000 0x0 0x200>;
+		pinctrl-0 = <&pinctrl0_usb &pinctrl1_usb>;
+		pinctrl-names = "default";
 		phys = <&saf1508_phy>;
 	};
 
@@ -54,14 +56,22 @@
 };
 
 &usdhc0 {
+	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-1 = <>;
+	pinctrl-2 = <>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	status = "okay";
 };
 
 &dspi1 {
+	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &dspi5 {
+	pinctrl-0 = <&pinctrl0_dspi5 &pinctrl1_dspi5>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
@@ -103,24 +113,32 @@
 /* EEPROM */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c0>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* Platform board PCI X16 EXPRESS - I2C_SCL_S0, I2C_SDA_S0  */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* Platform board GPIO_J3-17 (SDA), GPIO_J3-19 (SCL0)  */
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* PMIC */
 &i2c4 {
 	status = "okay";
+	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
+	pinctrl-names = "default";
 	clock-frequency=<100000>;
 
 	vr5510 {
@@ -136,3 +154,172 @@
 	};
 
 };
+
+&pinctrl0 {
+	board_pinctrl0 {
+		pinctrl0_i2c0: pinctrl0_i2c0 {
+			fsl,pins = <PB00_MSCR_S32G2XX PB00_I2C0_SDA_CFG
+				    PB01_MSCR_S32G2XX PB01_I2C0_SCL_CFG
+				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
+				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
+				    >;
+		};
+
+		pinctrl0_i2c1: pinctrl0_i2c1 {
+			fsl,pins = <PB03_MSCR_S32G2XX PB03_I2C1_SCL_CFG
+				    PB04_MSCR_S32G2XX PB04_I2C1_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_i2c2: pinctrl0_i2c2 {
+			fsl,pins = <PB05_MSCR_S32G2XX PB05_I2C2_SCL_CFG
+				    PB06_MSCR_S32G2XX PB06_I2C2_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_i2c4: pinctrl0_i2c4 {
+			fsl,pins = <PC01_MSCR_S32G2XX PC01_I2C4_SDA_CFG
+				    PC02_MSCR_S32G2XX PC02_I2C4_SCL_CFG
+				    >;
+		};
+
+		pinctrl0_qspi: pinctrl0_qspi {
+			fsl,pins = <PF05_MSCR_S32G2XX PF05_QSPI_DATA_A0_CFG
+				    PF06_MSCR_S32G2XX PF06_QSPI_DATA_A1_CFG
+				    PF07_MSCR_S32G2XX PF07_QSPI_DATA_A2_CFG
+				    PF08_MSCR_S32G2XX PF08_QSPI_DATA_A3_CFG
+				    PF09_MSCR_S32G2XX PF09_QSPI_DATA_A4_CFG
+				    PF10_MSCR_S32G2XX PF10_QSPI_DATA_A5_CFG
+				    PF11_MSCR_S32G2XX PF11_QSPI_DATA_A6_CFG
+				    PF12_MSCR_S32G2XX PF12_QSPI_DATA_A7_CFG
+				    PF13_MSCR_S32G2XX PF13_QSPI_DQS_A_CFG
+				    PG00_MSCR_S32G2XX PG00_QSPI_CLK_A_CFG
+				    PG01_MSCR_S32G2XX PG01_QSPI_CLK_A_b_CFG
+				    PG02_MSCR_S32G2XX PG02_QSPI_CLK_2A_CFG
+				    PG03_MSCR_S32G2XX PG03_QSPI_CLK_2A_b_CFG
+				    PG04_MSCR_S32G2XX PG04_QSPI_CS_A0
+				    PG05_MSCR_S32G2XX PG05_QSPI_CS_A1
+				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
+				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
+				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
+				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
+				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
+				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
+				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
+				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
+				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
+				    >;
+		};
+
+		pinctrl0_sd0: pinctrl0_sd0 {
+			fsl,pins = <PC14_MSCR_S32G2XX PC14_SD0_CLK_CFG
+				    PC15_MSCR_S32G2XX PC15_SD0_CMD_CFG
+				    PD00_MSCR_S32G2XX PD00_SD0_D0_CFG
+				    PD01_MSCR_S32G2XX PD01_SD0_D1_CFG
+				    PD02_MSCR_S32G2XX PD02_SD0_D2_CFG
+				    PD03_MSCR_S32G2XX PD03_SD0_D3_CFG
+				    PD04_MSCR_S32G2XX PD04_SD0_D4_CFG
+				    PD05_MSCR_S32G2XX PD05_SD0_D5_CFG
+				    PD06_MSCR_S32G2XX PD06_SD0_D6_CFG
+				    PD07_MSCR_S32G2XX PD07_SD0_D7_CFG
+				    PD08_MSCR_S32G2XX PD08_SD0_RST_CFG
+				    PD09_MSCR_S32G2XX PD09_SD0_VSELECT_CFG
+				    PD10_MSCR_S32G2XX PD10_SD0_DQS_CFG
+				    SD0_CMD_IMCR PC15_SD0_CMD_IN
+				    SD0_D0_IMCR PD00_SD0_D0_IN
+				    SD0_D1_IMCR PD01_SD0_D1_IN
+				    SD0_D2_IMCR PD02_SD0_D2_IN
+				    SD0_D3_IMCR PD03_SD0_D3_IN
+				    SD0_D4_IMCR PD04_SD0_D4_IN
+				    SD0_D5_IMCR PD05_SD0_D5_IN
+				    SD0_D6_IMCR PD06_SD0_D6_IN
+				    SD0_D7_IMCR PD07_SD0_D7_IN
+				    SD0_DQS_IMCR PD10_SD0_DQS_IN
+				    >;
+		};
+
+		pinctrl0_dspi1: pinctrl0_dspi1 {
+			fsl,pins = <PA06_MSCR_S32G2XX PA06_SPI1_SOUT_CFG
+				    PA07_MSCR_S32G2XX PA07_SPI1_CS0_CFG
+				    PA08_MSCR_S32G2XX PA08_SPI1_SCK_CFG
+				    PF15_MSCR_S32G2XX PF15_SPI1_SIN_CFG
+				    >;
+		};
+
+		pinctrl0_dspi5: pinctrl0_dspi5 {
+			fsl,pins = <PA09_MSCR_S32G2XX PA09_SPI5_SCK_CFG
+				    PA10_MSCR_S32G2XX PA10_SPI5_SIN_CFG
+				    PA11_MSCR_S32G2XX PA11_SPI5_SOUT_CFG
+				    PA12_MSCR_S32G2XX PA12_SPI5_CS0_CFG
+				    >;
+		};
+
+		pinctrl0_usb: pinctrl0_usb {
+			fsl,pins = <PD14_MSCR_S32G2XX PD14_USB_DATA0_CFG
+				    PD15_MSCR_S32G2XX PD15_USB_DATA1_CFG
+				    PE00_MSCR_S32G2XX PE00_USB_DATA2_CFG
+				    PE01_MSCR_S32G2XX PE01_USB_DATA3_CFG
+				    >;
+		};
+	};
+};
+
+&pinctrl1 {
+	board_pinctrl1 {
+		pinctrl1_i2c1: pinctrl1_i2c1 {
+			fsl,pins = <I2C1_SCL_IMCR PB03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c2: pinctrl1_i2c2 {
+			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c4: pinctrl1_i2c4 {
+			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
+				    >;
+		};
+
+		pinctrl1_dspi1: pinctrl1_dspi1 {
+			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi5: pinctrl1_dspi5 {
+			fsl,pins = <DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
+				    >;
+		};
+
+		pinctrl1_usb: pinctrl1_usb {
+			fsl,pins = <PH00_MSCR_S32G2XX PH00_USB_DATA7_CFG
+				    PL08_MSCR_S32G2XX PL08_USB_CLK_CFG
+				    PL09_MSCR_S32G2XX PL09_USB_DIR_CFG
+				    PL10_MSCR_S32G2XX PL10_USB_STP_CFG
+				    PL11_MSCR_S32G2XX PL11_USB_NXT_CFG
+				    PL12_MSCR_S32G2XX PL12_USB_DATA4_CFG
+				    PL13_MSCR_S32G2XX PL13_USB_DATA5_CFG
+				    PL14_MSCR_S32G2XX PL14_USB_DATA6_CFG
+				    USB_ULPI_CLK_IMCR PL08_USB_CLK_IN
+				    USB_ULPI_DATA0_IMCR PD14_USB_DATA0_IN
+				    USB_ULPI_DATA1_IMCR PD15_USB_DATA1_IN
+				    USB_ULPI_DATA2_IMCR PE00_USB_DATA2_IN
+				    USB_ULPI_DATA3_IMCR PE01_USB_DATA3_IN
+				    USB_ULPI_DATA4_IMCR PL12_USB_DATA4_IN
+				    USB_ULPI_DATA5_IMCR PL13_USB_DATA5_IN
+				    USB_ULPI_DATA6_IMCR PL14_USB_DATA6_IN
+				    USB_ULPI_DATA7_IMCR PH00_USB_DATA7_IN
+				    USB_ULPI_DIR_IMCR PL09_USB_DIR_IN
+				    USB_ULPI_NXT_IMCR PL11_USB_NXT_IN
+				    >;
+		};
+	};
+};
+
+&qspi {
+	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-names = "default";
+};
diff --git a/arch/arm/dts/fsl-s32g274ardb.dtsi b/arch/arm/dts/fsl-s32g274ardb.dtsi
index 6edbc978aa..8ed22e6bc6 100644
--- a/arch/arm/dts/fsl-s32g274ardb.dtsi
+++ b/arch/arm/dts/fsl-s32g274ardb.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 /*
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  */
 
 /dts-v1/;
@@ -24,14 +24,23 @@
 };
 
 &usdhc0 {
+	/* By default sd0 pins were able to work at 100Mhz and 200Mhz */
+	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-1 = <>;
+	pinctrl-2 = <>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	status = "okay";
 };
 
 &dspi1 {
+	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 &dspi5 {
+	pinctrl-0 = <&pinctrl0_dspi5 &pinctrl1_dspi5>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
@@ -42,24 +51,32 @@
 /* IO Expander  */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c0>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* PCIe X1 Connector  */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl1_i2c1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* FLEXRAY_LIN  */
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* PMIC */
 &i2c4 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	vr5510 {
@@ -74,3 +91,137 @@
 		status = "okay";
 	};
 };
+
+&pinctrl0 {
+	board_pinctrl0 {
+		pinctrl0_i2c0: pinctrl0_i2c0 {
+			fsl,pins = <PB15_MSCR_S32G2XX PB15_I2C0_SDA_CFG
+				    PC00_MSCR_S32G2XX PC00_I2C0_SCL_CFG
+				    I2C0_SDA_IMCR PB15_I2C0_SDA_IN
+				    I2C0_SCL_IMCR PC00_I2C0_SCL_IN
+				    >;
+		};
+
+		pinctrl0_i2c2: pinctrl0_i2c2 {
+			fsl,pins = <PB05_MSCR_S32G2XX PB05_I2C2_SCL_CFG
+				    PB06_MSCR_S32G2XX PB06_I2C2_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_i2c4: pinctrl0_i2c4 {
+			fsl,pins = <PC01_MSCR_S32G2XX PC01_I2C4_SDA_CFG
+				    PC02_MSCR_S32G2XX PC02_I2C4_SCL_CFG
+				    >;
+		};
+
+		pinctrl0_qspi: pinctrl0_qspi {
+			fsl,pins = <PF05_MSCR_S32G2XX PF05_QSPI_DATA_A0_CFG
+				    PF06_MSCR_S32G2XX PF06_QSPI_DATA_A1_CFG
+				    PF07_MSCR_S32G2XX PF07_QSPI_DATA_A2_CFG
+				    PF08_MSCR_S32G2XX PF08_QSPI_DATA_A3_CFG
+				    PF09_MSCR_S32G2XX PF09_QSPI_DATA_A4_CFG
+				    PF10_MSCR_S32G2XX PF10_QSPI_DATA_A5_CFG
+				    PF11_MSCR_S32G2XX PF11_QSPI_DATA_A6_CFG
+				    PF12_MSCR_S32G2XX PF12_QSPI_DATA_A7_CFG
+				    PF13_MSCR_S32G2XX PF13_QSPI_DQS_A_CFG
+				    PG00_MSCR_S32G2XX PG00_QSPI_CLK_A_CFG
+				    PG01_MSCR_S32G2XX PG01_QSPI_CLK_A_b_CFG
+				    PG02_MSCR_S32G2XX PG02_QSPI_CLK_2A_CFG
+				    PG03_MSCR_S32G2XX PG03_QSPI_CLK_2A_b_CFG
+				    PG04_MSCR_S32G2XX PG04_QSPI_CS_A0
+				    PG05_MSCR_S32G2XX PG05_QSPI_CS_A1
+				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
+				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
+				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
+				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
+				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
+				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
+				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
+				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
+				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
+				    >;
+		};
+
+		pinctrl0_sd0: pinctrl0_sd0 {
+			fsl,pins = <PC14_MSCR_S32G2XX PC14_SD0_CLK_CFG
+				    PC15_MSCR_S32G2XX PC15_SD0_CMD_CFG
+				    PD00_MSCR_S32G2XX PD00_SD0_D0_CFG
+				    PD01_MSCR_S32G2XX PD01_SD0_D1_CFG
+				    PD02_MSCR_S32G2XX PD02_SD0_D2_CFG
+				    PD03_MSCR_S32G2XX PD03_SD0_D3_CFG
+				    PD04_MSCR_S32G2XX PD04_SD0_D4_CFG
+				    PD05_MSCR_S32G2XX PD05_SD0_D5_CFG
+				    PD06_MSCR_S32G2XX PD06_SD0_D6_CFG
+				    PD07_MSCR_S32G2XX PD07_SD0_D7_CFG
+				    PD08_MSCR_S32G2XX PD08_SD0_RST_CFG
+				    PD09_MSCR_S32G2XX PD09_SD0_VSELECT_CFG
+				    PD10_MSCR_S32G2XX PD10_SD0_DQS_CFG
+				    SD0_CMD_IMCR PC15_SD0_CMD_IN
+				    SD0_D0_IMCR PD00_SD0_D0_IN
+				    SD0_D1_IMCR PD01_SD0_D1_IN
+				    SD0_D2_IMCR PD02_SD0_D2_IN
+				    SD0_D3_IMCR PD03_SD0_D3_IN
+				    SD0_D4_IMCR PD04_SD0_D4_IN
+				    SD0_D5_IMCR PD05_SD0_D5_IN
+				    SD0_D6_IMCR PD06_SD0_D6_IN
+				    SD0_D7_IMCR PD07_SD0_D7_IN
+				    SD0_DQS_IMCR PD10_SD0_DQS_IN
+				    >;
+		};
+
+		pinctrl0_dspi1: pinctrl0_dspi1 {
+			fsl,pins = <PA06_MSCR_S32G2XX PA06_SPI1_SOUT_CFG
+				    PA07_MSCR_S32G2XX PA07_SPI1_CS0_CFG
+				    PA08_MSCR_S32G2XX PA08_SPI1_SCK_CFG
+				    PF15_MSCR_S32G2XX PF15_SPI1_SIN_CFG
+				    >;
+		};
+
+		pinctrl0_dspi5: pinctrl0_dspi5 {
+			fsl,pins = <PA09_MSCR_S32G2XX PA09_SPI5_SCK_CFG
+				    PA10_MSCR_S32G2XX PA10_SPI5_SIN_CFG
+				    PA11_MSCR_S32G2XX PA11_SPI5_SOUT_CFG
+				    PA12_MSCR_S32G2XX PA12_SPI5_CS0_CFG
+				    >;
+		};
+	};
+};
+
+&pinctrl1 {
+	board_pinctrl1 {
+		pinctrl1_i2c1: pinctrl1_i2c1 {
+			fsl,pins = <PK03_MSCR_S32G2XX PK03_I2C1_SCL_CFG
+				    PK05_MSCR_S32G2XX PK05_I2C1_SDA_CFG
+				    I2C1_SCL_IMCR PK03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PK05_I2C1_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c2: pinctrl1_i2c2 {
+			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
+				    >;
+		};
+
+		pinctrl1_i2c4: pinctrl1_i2c4 {
+			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
+				    >;
+		};
+
+		pinctrl1_dspi1: pinctrl1_dspi1 {
+			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi5: pinctrl1_dspi5 {
+			fsl,pins = <DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
+				    >;
+		};
+	};
+};
+
+&qspi {
+	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-names = "default";
+};
diff --git a/arch/arm/dts/fsl-s32r45.dts b/arch/arm/dts/fsl-s32r45.dts
index 5362abf8a5..c04094f5ac 100644
--- a/arch/arm/dts/fsl-s32r45.dts
+++ b/arch/arm/dts/fsl-s32r45.dts
@@ -9,6 +9,7 @@
  */
 #include <dt-bindings/clock/s32r45-clock.h>
 #include <dt-bindings/clock/s32r45-scmi-clock.h>
+#include <dt-bindings/pinctrl/s32r45-pinctrl.h>
 
 /dts-v1/;
 #include "fsl-s32-gen1.dtsi"
@@ -166,6 +167,8 @@
 };
 
 &qspi {
+	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	mx25uw51245g@0 {
@@ -180,6 +183,8 @@
 };
 
 &usdhc0 {
+	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
@@ -200,6 +205,9 @@
 	clock-names = "rx_sgmii", "tx_sgmii", "ts_sgmii",
 		      "rx_rgmii", "tx_rgmii", "ts_rgmii",
 		      "axi";
+	pinctrl-0 = <&pinctrl0_gmac0 &pinctrl0_gmac0_mdio>;
+	pinctrl-1 = <&pinctrl0_gmac0_mdio>;
+	pinctrl-names = "gmac_rgmii", "gmac_sgmii";
 	status = "okay";
 	phy-mode = "rgmii";
 	phy-handle = <&gmac0_phy1>;
@@ -217,12 +225,201 @@
 /* J36 - PMIC */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c0>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
 /* J37 */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
+	pinctrl-names = "default";
 	status = "okay";
 };
 
+&pinctrl0 {
+	board_pinctrl0 {
+		pinctrl0_dspi1: pinctrl0_dspi1 {
+			fsl,pins = <PB04_MSCR_S32R45 PB04_SPI1_SCK_CFG
+				    PB05_MSCR_S32R45 PB05_SPI1_SIN_CFG
+				    PB06_MSCR_S32R45 PB06_SPI1_SOUT_CFG
+				    PB07_MSCR_S32R45 PB07_SPI1_CS0_CFG
+				    PB08_MSCR_S32R45 PB08_SPI1_CS1_CFG
+				    >;
+		};
+
+		pinctrl0_dspi2: pinctrl0_dspi2 {
+			fsl,pins = <PB11_MSCR_S32R45 PB11_SPI2_SCK_CFG
+				    PB13_MSCR_S32R45 PB13_SPI2_SOUT_CFG
+				    PB14_MSCR_S32R45 PB14_SPI2_CS0_CFG
+				    PB15_MSCR_S32R45 PB15_SPI2_CS1_CFG
+				    PC01_MSCR_S32R45 PC01_SPI2_SIN_CFG
+				    >;
+		};
+
+		pinctrl0_dspi3: pinctrl0_dspi3 {
+			fsl,pins = <PC04_MSCR_S32R45 PC04_SPI3_SCK_CFG
+				    PC06_MSCR_S32R45 PC06_SPI3_SIN_CFG
+				    PC07_MSCR_S32R45 PC07_SPI3_CS0_CFG
+				    PC08_MSCR_S32R45 PC08_SPI3_CS1_CFG
+				    PC13_MSCR_S32R45 PC13_SPI3_SOUT_CFG
+				    >;
+		};
+
+		pinctrl0_gmac0_mdio: pinctrl0_gmac0_mdio {
+			fsl,pins = <PD12_MSCR_S32R45 PD12_GMAC0_MDC_CFG
+				    PD13_MSCR_S32R45 PD13_GMAC0_MDIO_CFG
+				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
+				    >;
+		};
+
+		pinctrl0_gmac0: pinctrl0_gmac0 {
+			fsl,pins = <PE02_MSCR_S32R45 PE02_GMAC0_TX_CLK_CFG
+				    PE03_MSCR_S32R45 PE03_GMAC0_TX_EN_CFG
+				    PE04_MSCR_S32R45 PE04_GMAC0_TX_D0_CFG
+				    PE05_MSCR_S32R45 PE05_GMAC0_TX_D1_CFG
+				    PE06_MSCR_S32R45 PE06_GMAC0_TX_D2_CFG
+				    PE07_MSCR_S32R45 PE07_GMAC0_TX_D3_CFG
+				    PE08_MSCR_S32R45 PE08_GMAC0_RX_CLK_CFG
+				    PE09_MSCR_S32R45 PE09_GMAC0_RX_DV_CFG
+				    PE10_MSCR_S32R45 PE10_GMAC0_RX_D0_CFG
+				    PE11_MSCR_S32R45 PE11_GMAC0_RX_D1_CFG
+				    PE12_MSCR_S32R45 PE12_GMAC0_RX_D2_CFG
+				    PE13_MSCR_S32R45 PE13_GMAC0_RX_D3_CFG
+				    GMAC0_TX_CLK_IMCR PE02_GMAC0_TX_CLK_IN
+				    GMAC0_RX_CLK_IMCR PE08_GMAC0_RX_CLK_IN
+				    GMAC0_RX_DV_IMCR PE09_GMAC0_RX_DV_IN
+				    GMAC0_RX_D0_IMCR PE10_GMAC0_RX_D0_IN
+				    GMAC0_RX_D1_IMCR PE11_GMAC0_RX_D1_IN
+				    GMAC0_RX_D2_IMCR PE12_GMAC0_RX_D2_IN
+				    GMAC0_RX_D3_IMCR PE13_GMAC0_RX_D3_IN
+				    >;
+		};
+
+		pinctrl0_i2c0: pinctrl0_i2c0 {
+			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_CFG
+				    PB01_MSCR_S32R45 PB01_I2C0_SCL_CFG
+				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
+				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
+				    >;
+		};
+
+		pinctrl0_i2c1: pinctrl0_i2c1 {
+			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_CFG
+				    PA15_MSCR_S32R45 PA15_I2C1_SDA_CFG
+				    >;
+		};
+
+		pinctrl0_qspi: pinctrl0_qspi {
+			fsl,pins = <PF05_MSCR_S32R45 PF05_QSPI_DATA_A0_CFG
+				    PF06_MSCR_S32R45 PF06_QSPI_DATA_A1_CFG
+				    PF07_MSCR_S32R45 PF07_QSPI_DATA_A2_CFG
+				    PF08_MSCR_S32R45 PF08_QSPI_DATA_A3_CFG
+				    PF09_MSCR_S32R45 PF09_QSPI_DATA_A4_CFG
+				    PF10_MSCR_S32R45 PF10_QSPI_DATA_A5_CFG
+				    PF11_MSCR_S32R45 PF11_QSPI_DATA_A6_CFG
+				    PF12_MSCR_S32R45 PF12_QSPI_DATA_A7_CFG
+				    PF13_MSCR_S32R45 PF13_QSPI_DQS_A_CFG
+				    PG00_MSCR_S32R45 PG00_QSPI_CLK_A_CFG
+				    PG01_MSCR_S32R45 PG01_QSPI_CLK_A_b_CFG
+				    PG02_MSCR_S32R45 PG02_QSPI_CLK_2A_CFG
+				    PG03_MSCR_S32R45 PG03_QSPI_CLK_2A_b_CFG
+				    PG04_MSCR_S32R45 PG04_QSPI_CS_A0
+				    PG05_MSCR_S32R45 PG05_QSPI_CS_A1
+				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
+				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
+				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
+				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
+				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
+				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
+				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
+				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
+				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
+				    >;
+		};
+
+		pinctrl0_sd0: pinctrl0_sd0 {
+			fsl,pins = <PC14_MSCR_S32R45 PC14_SD0_CLK_CFG
+				    PC15_MSCR_S32R45 PC15_SD0_CMD_CFG
+				    PD00_MSCR_S32R45 PD00_SD0_D0_CFG
+				    PD01_MSCR_S32R45 PD01_SD0_D1_CFG
+				    PD02_MSCR_S32R45 PD02_SD0_D2_CFG
+				    PD03_MSCR_S32R45 PD03_SD0_D3_CFG
+				    PD04_MSCR_S32R45 PD04_SD0_D4_CFG
+				    PD05_MSCR_S32R45 PD05_SD0_D5_CFG
+				    PD06_MSCR_S32R45 PD06_SD0_D6_CFG
+				    PD07_MSCR_S32R45 PD07_SD0_D7_CFG
+				    PD08_MSCR_S32R45 PD08_SD0_RST_CFG
+				    PD09_MSCR_S32R45 PD09_SD0_VSELECT_CFG
+				    SD0_CMD_IMCR PC15_SD0_CMD_IN
+				    SD0_D0_IMCR PD00_SD0_D0_IN
+				    SD0_D1_IMCR PD01_SD0_D1_IN
+				    SD0_D2_IMCR PD02_SD0_D2_IN
+				    SD0_D3_IMCR PD03_SD0_D3_IN
+				    SD0_D4_IMCR PD04_SD0_D4_IN
+				    SD0_D5_IMCR PD05_SD0_D5_IN
+				    SD0_D6_IMCR PD06_SD0_D6_IN
+				    SD0_D7_IMCR PD07_SD0_D7_IN
+				    >;
+		};
+	};
+};
+
+&pinctrl1 {
+	board_pinctrl1 {
+		pinctrl1_dspi1: pinctrl1_dspi1 {
+			fsl,pins = <DSPI1_SIN_IMCR PB05_SPI1_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi2: pinctrl1_dspi2 {
+			fsl,pins = <DSPI2_SIN_IMCR PC01_SPI2_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi3: pinctrl1_dspi3 {
+			fsl,pins = <DSPI3_SIN_IMCR PC06_SPI3_SIN_IN
+				    >;
+		};
+
+		pinctrl1_dspi5: pinctrl1_dspi5 {
+			fsl,pins = <PK00_MSCR_S32R45 PK00_SPI5_SCK_CFG
+				    PK03_MSCR_S32R45 PK03_SPI5_CS0_CFG
+				    PK04_MSCR_S32R45 PK04_SPI5_SIN_CFG
+				    PK05_MSCR_S32R45 PK05_SPI5_SOUT_CFG
+				    DSPI5_SIN_IMCR PK04_SPI5_SIN_IN
+				    >;
+		};
+
+		pinctrl1_i2c1: pinctrl1_i2c1 {
+			fsl,pins = <I2C1_SCL_IMCR PA14_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PA15_I2C1_SDA_IN
+				    >;
+		};
+	};
+};
+
+&dspi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_dspi1>, <&pinctrl1_dspi1>;
+	status = "okay";
+};
+
+&dspi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_dspi2>, <&pinctrl1_dspi2>;
+	status = "okay";
+};
+
+&dspi3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_dspi3>, <&pinctrl1_dspi3>;
+	status = "okay";
+};
+
+&dspi5 {
+	pinctrl-names = "default";
+	pinctrl-0 =  <&pinctrl1_dspi5>;
+	status = "okay";
+};
-- 
2.17.1

